src/dashboard/src/components/archival_storage/atom.py

```diff
@@ -0,0 +1,104 @@
+# This file is part of Archivematica.
+#
+# Copyright 2010-2016 Artefactual Systems Inc. <http://artefactual.com>
+#
+# Archivematica is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Archivematica is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Archivematica.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import os
+import tempfile
+
+from agentarchives.atom.client import AtomClient, AtomError
+from metsrw import METSDocument, MetsError
+from storageService import extract_file
+from slumber.exceptions import SlumberHttpBaseException
+
+from main.models import DashboardSetting
+
+
+logger = logging.getLogger('archivematica.dashboard')
+
+
+def get_atom_client():
+    url = DashboardSetting.objects.get(name='dip_upload_atom_url').value
+    key = DashboardSetting.objects.get(name='dip_upload_atom_key').value
+    return AtomClient(url, key)
+
+
+class AtomMetadataUploadError(Exception):
+    pass
+
+
+def upload_dip_metadata_to_atom(aip_name, aip_uuid, parent_slug):
+    """
+    Write to a AtoM's resource (parent_slug) the metadata of the objects of a
+    AIP given its name and UUID. Return the slug of the new container resource
+    created to hold the metadata objects.
+    """
+    try:
+        # Download METS file
+        _, temp_path = tempfile.mkstemp()
```

Comment:
[TemporaryFile](https://docs.python.org/2/library/tempfile.html#tempfile.TemporaryFile) can be used with a `with` statement, which will automatically close itself so you don't have to do the try/except/finally.  This also means the exception handling can be around the smallest part of the code likely to generate the exception.